### DISCUSSION TITLE
[WIP] Supported Features

### DISCUSSION THREAD

**Discussion by GeorgeS2019** [V-Sekai/unidot_importer#16] [General]
### Update 16th May, use Godot4.0.3

- [x] Unity Prefabs to Godot Scenes 
- [ ] Compress textures => fbx model imported without texture
- [ ] A system will map equivalent Godot shaders
- [ ] file1Animation.fbx
- [ ] file2Animation.anim <= not yet

**Comment by lyuma** [comment]
Can you explain what "Compress textures => fbx model imported without texture" means? I think this should work. What issues did you run into?

 file1Animation.fbx Should be working now. Though there are still known bugs with humanoid animations, especially for cases where non-human child bones or meshes are animated too.

**Reply by GeorgeS2019** [reply]
Come from this issue:
https://github.com/V-Sekai/unidot_importer/issues/2

When I export as e.g. unitypackages or fbx, the textures are not attached. 
Export as gltf works

We write here so others could try to keep updating the experience

**Reply by lyuma** [reply]
Can you share more details? This is supposed to work.

If you have a unitypackage that reproduces the problem and would be willing to send me a DM, that might be ok. otherwise, could you share the output log? (select all from console panel or pipe to a file)

**Comment by GeorgeS2019** [comment]
FYI: Alternative Unitypackage [importer ](https://www.reddit.com/r/godot/comments/13io0bx/load_unitypackage/) for godot
> My main goal for this is to handle most of the conversion process for all the Synty assets I purchased over the years before giving up on Unity.
![msedge_6Y4F0JrS1Y](https://github.com/V-Sekai/unidot_importer/assets/49812372/f33c4e02-66a3-4d9a-aace-9c0bc782472d)

**Reply by GeorgeS2019** [reply]
From the author:

I wrote a command line utility in **Rust** to interface with .unitypackage files. 

- It handles the tar.gz compression, extraction of files, 
- translation from **packed naming convention** to **disk naming convention**, 
- conversion of yaml to json, and 
- processing fbx to glb using a **modified** fbx2gltf to keep **transform origin.** => [Some models are off by 90 degrees](https://github.com/V-Sekai/unidot_importer/issues/10)
   > A GLTFDocumentExtension written in GDScript assists with keeping the correct transform origin. 

- The rest of the processing is done in GDScript. 
  > - It walks the yaml to json converted asset meta data and 
  > - rebuilds each transform, saving prefabs/materials/meshes along the way. 
  > - It also handles the left-to-right hand conversions.

- If it detects a standard unity material it recreates what it can (color, texture, alpha).
- It can load full *.unity scene files, with the prefabs, included nested prefabs with child attribute overrides.

> My main goal for this is to handle most of the conversion process for all the Synty assets I purchased over the years before giving up on Unity.

**Reply by lyuma** [reply]
Yes, the person who made this post released a video but to my knowledge hasn't released source code to it. I'm looking forward to seeing what end product they put out.

Almost all of the above features are already present in Unidot.

The one I'm unsure about is:
> modified fbx2gltf to keep transform origin. 

There is a possible issue for some FBX files that use its RotationPivot feature to offset meshes. There was an attempt to patch this in, but I'm not sure about the current state. The author did not release any source code, so I cannot evaluate what they changed.
